KernelCommunicator.swift

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * A struct with functions that the kernel and the code running inside the
+ * kernel use to talk to each other.
+ *
+ * Note that it would be more Juptyer-y for the communication to happen over
+ * ZeroMQ. This is not currently possible, because ZeroMQ sends messages
+ * asynchronously using IO threads, and LLDB pauses those IO threads, which
+ * prevents them from sending the messages.
+ */
+public struct KernelCommunicator {
+  private var afterSuccessfulExecutionHandlers: [() -> JuptyerMessages]
+  private var parentMessageHandlers: [(ParentMessage) -> ()]
+
+  public let juptyerSession: JuptyerSession
```

Comment:
Typo: `juptyerSession` -> `jupyterSession`
"Jupyter" is misspelled in other places too (`JuptyerDisplayMessage`, `JuptyerMessages`, etc).

Reply:
wow I did that a lot. fixed!